MAINTENANCE.md

```diff
@@ -0,0 +1,144 @@
+compatibility, but they can also be cut more frequently as desired. Here
+is the recommended process:
+
+- Check that all repos are up to date with cherry picks, and that all
+  open cherry pick PRs have been merged
+- Lightly triage open PRs and issues to make sure that anything that
+  should be merged or fixed before the new release has been
+- Compare the current HEAD of the repo with the last release using a
+  command such as `git log 0.21.1...HEAD` or [on
+  GitHub](https://github.com/bazelbuild/rules_apple/compare/0.21.1...HEAD)
+  and collect the most notable user facing commits for the release
+  notes
+- Starting with `apple_support` create a new release with this template
+  for the notes:
+
+```
+- NOTABLE CHANGE 1
+- NOTABLE CHANGE 2
+- This release is tested with Bazel N.N.N
+
+Please use the release asset from your Bazel WORKSPACE instead of
+GitHub's asset to reduce download size and improve reproducibility.
+
+SHA-256 digest: `TBD`
+```
+
+- Download the source archive GitHub produced with the release,
+  unarchive it, and create a release archive with this command: `tar
+  czvf REPO_NAME.RELEASE_VERSION.tar.gz *` (ideally this would be
+  scripted in the future and more artifacts would be excluded to reduce
+  download size, but this method avoids potentially gitignored artifacts
+  in the archives).
+- Update the release with the archive, and update the sha256 with the
+  output of `shasum -a 256 ARCHIVE`
+- Update the `apple_support` `README.md` with the new version and sha256
+  to make it easier for users to copy and paste to their `WORKSPACE`
+- Update the `swift/repositories.bzl` file in `rules_swift` with the new
+  `apple_support` release
+- Repeat the steps above to create a release on `rules_swift`
+- Update the `rules_swift` `README.md` with the new version and sha256
+  to make it easier for users to copy and paste to their `WORKSPACE`
+- Update the `apple/repositories.bzl` file in `rules_apple` with the new
+  `apple_support` and `rules_swift` releases
+- Repeat the steps above to create a release on `rules_apple`
+- Update the `rules_apple` `README.md` with the new version and sha256
+  to make it easier for users to copy and paste to their `WORKSPACE`
+- Update the `Tulsi` `WORKSPACE` with the new version of `rules_apple`
+
+### Notes
+
+- The rules aren't currently following true semantic versioning, but in
+  general the minor version should be bumped for most changes, and the
+  patch only for very small releases
+- It's highly recommended that rules maintainers track more closely with
+  the HEAD of the rules repos than with the releases
+- In general before releasing the HEAD of the rules repos should be
+  tested on non-trivial projects, ideally that is mostly covered by the
+  point above
+- Not all rules repos will have changes every time you go to create a
+  new release, in those cases you can skip those repos and the version
+  bumps associated with them
```

Comment:
Do we want to say something about updating the release message to include additional versions of Bazel that were tested with it? That would help answer questions of "is this release up to date, or did the maintainers forget to cut a new release"?

Reply:
added